code/modules/cooking/machinery/commissary.dm

```diff
@@ -0,0 +1,363 @@
+	var/item = user.get_active_hand()
+	var/obj/item/card/id/I = item
+	if(istype(I) && (ACCESS_HEADS in I.access))
+		if(ACCESS_HEADS in I.access)
+			editmode = TRUE
+			to_chat(user, SPAN_NOTICE("Command access granted."))
+			SStgui.update_uis(src)
+		return
+	if(istype(item, /obj/item/spacecash) && !istype(item, /obj/item/spacecash/ewallet))
+		var/obj/item/spacecash/cashmoney = item
+		credit += cashmoney.worth
+		user.drop_from_inventory(cashmoney,get_turf(src))
+		visible_message("\The [user] inserts some credits into \the [src]." )
+		qdel(cashmoney)
+		return
+	I = user.GetIdCard()
+	if(istype(I) && (ACCESS_CARGO in I.access))
+		var/price_guess = text2num(sanitizeSafe( tgui_input_text(user, "How much do you wish to withdraw? Remaining cash: [credit]", "QuikPay", 0, 10), 10))
+		if(isnull(price_guess) || price_guess == 0)
+			return
+		price_guess = max(0, round(price_guess, 0.01))
+		if(credit >= price_guess)
+			spawn_money(price_guess, loc, user)
+			credit -= price_guess
+		visible_message("\The [user] remove some credits from \the [src]." )
+		return
+
+/obj/structure/cash_register/commissary/proc/print_receipt()
+	var/obj/item/paper/R = new(loc)
+	var/receiptname = "Receipt: [machine_id]"
+	R.set_content_unsafe(receiptname, receipt, sum)
+
+	//stamp the paper
+	var/image/stampoverlay = image('icons/obj/bureaucracy.dmi')
+	stampoverlay.icon_state = "paper_stamp-cent"
+	if(!R.stamped)
+		R.stamped = new
+	R.stamped += /obj/item/stamp
+	R.AddOverlays(stampoverlay)
+	R.stamps += "<HR><i>This paper has been stamped by the Idris Quik-Pay Register.</i>"
+	usr.put_in_any_hand_if_possible(R)
+
+/obj/structure/cash_register/commissary/attackby(obj/item/attacking_item, mob/user)
+	if(sum == 0)
+		return
+	if (istype(attacking_item, /obj/item/spacecash/ewallet))
+		card_pay(attacking_item, user)
+		return
+	else if (istype(attacking_item, /obj/item/card/id))
+		ID_pay(attacking_item, user)
+		return
+	else if(istype(attacking_item, /obj/item/spacecash))
+		cash_pay(attacking_item, user)
+		return
+
+/obj/structure/cash_register/commissary/proc/cash_pay(obj/item/spacecash/cashmoney, mob/user)
+	var/transaction_amount = sum
+	if(transaction_amount > cashmoney.worth)
+		to_chat(user, SPAN_WARNING("[icon2html(cashmoney, user)] That is not enough money."))
+		return 0
+	if(istype(cashmoney, /obj/item/spacecash/bundle))
+		visible_message(SPAN_INFO("\The [user] inserts some cash into \the [src]."))
+		var/obj/item/spacecash/bundle/cashmoney_bundle = cashmoney
+		cashmoney_bundle.worth -= transaction_amount
+
+		if(cashmoney_bundle.worth <= 0)
+			usr.drop_from_inventory(cashmoney_bundle,get_turf(src))
+			qdel(cashmoney_bundle)
+		else
+			cashmoney_bundle.update_icon()
+	else
+		visible_message(SPAN_INFO("\The [user] inserts a bill into \the [src]."))
+		var/left = cashmoney.worth - transaction_amount
+		user.drop_from_inventory(cashmoney,get_turf(src))
+		qdel(cashmoney)
+
+		if(left)
+			spawn_money(left, get_turf(user), user)
```

Comment:
IMHO you should try to put it into the active hand instead of dropping it on the floor.